examples/fwi_operators.py

```diff
@@ -109,25 +109,25 @@ def grid2point(self, u, pt_coords):
             subs.append((ry, y))
         subs.append((rz, z))
         if self.ndim == 2:
-            return sum([b.subs(subs) * u[t, i+inc[0]+self.nbpml, k+inc[1]+self.nbpml] for inc, b in zip(self.increments, self.bs)])
+            return sum([b.subs(subs) * u.indexed[t, i+inc[0]+self.nbpml, k+inc[1]+self.nbpml] for inc, b in zip(self.increments, self.bs)])
         else:
-            return sum([b.subs(subs) * u[t, i+inc[0]+self.nbpml, j+inc[1]+self.nbpml, k+inc[2]+self.nbpml] for inc, b in zip(self.increments, self.bs)])
+            return sum([b.subs(subs) * u.indexed[t, i+inc[0]+self.nbpml, j+inc[1]+self.nbpml, k+inc[2]+self.nbpml] for inc, b in zip(self.increments, self.bs)])
 
     def read(self, u):
         eqs = []
-        for i in range(self.npoints):
-            eqs.append(Eq(self[t, i], self.grid2point(u, self.orig_data[i, :])))
+        for i in range(self.npoint):
+            eqs.append(Eq(self.indexed[t, i], self.grid2point(u, self.orig_data[i, :])))
```

Comment:
Maybe, but to be honest, I don't really know how we should preoperly treat `PointData` symbolically yet. So for I'd like to stay explicit, until we have really figured out this all fits into the overall paradigm.